library/pkwrite.c

```diff
@@ -374,13 +496,54 @@ int mbedtls_pk_write_key_der(const mbedtls_pk_context *key, unsigned char *buf,
     int ret = MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED;
     unsigned char *c;
     size_t len = 0;
+#if defined(MBEDTLS_ECP_LIGHT)
+    int is_ec_opaque = 0;
+#if defined(MBEDTLS_PK_HAVE_RFC8410_CURVES)
+    int is_montgomery_opaque = 0;
+#endif /* MBEDTLS_PK_HAVE_RFC8410_CURVES */
+#endif /* MBEDTLS_ECP_LIGHT */
+#if defined(MBEDTLS_USE_PSA_CRYPTO)
+    psa_key_attributes_t opaque_key_attr = PSA_KEY_ATTRIBUTES_INIT;
+    psa_key_type_t opaque_key_type = PSA_KEY_TYPE_NONE;
+    uint8_t tmp[PSA_EXPORT_KEY_PAIR_MAX_SIZE];
+    size_t tmp_len = 0;
+#endif /* MBEDTLS_USE_PSA_CRYPTO */
 
     if (size == 0) {
         return MBEDTLS_ERR_ASN1_BUF_TOO_SMALL;
     }
 
     c = buf + size;
 
+#if defined(MBEDTLS_USE_PSA_CRYPTO)
+    if (mbedtls_pk_get_type(key) == MBEDTLS_PK_OPAQUE) {
+        if (psa_get_key_attributes(key->priv_id,
```

Comment:
`psa_reset_key_attributes` is not used in this function on error / success. Is this deliberate?

Reply:
No, that's an error, sorry.
However I wonder why the CI was happy even without `psa_reset_key_attributes`.  If `psa_get_key_attributes` allocates something that it's not free in the end, isn't sanity checker expected to complain?

Reply:
Yep, since the spec says that the function might allocate resources, the reset has to be called regardless of input.